transcription/ibm.go

```diff
@@ -0,0 +1,134 @@
+package transcription
+
+import (
+	"bufio"
+	"encoding/base64"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// IBMResult is the result of an IBM transcription. See
+// https://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/doc/speech-to-text/output.shtml
+// for details.
+type IBMResult struct {
+	ResultIndex int              `json:"result_index"`
+	Results     []ibmResultField `json:"results"`
+}
+type ibmResultField struct {
+	Alternatives []ibmAlternativesField `json:"alternatives"`
+	Final        bool                   `json:"final"`
+}
+type ibmAlternativesField struct {
+	WordConfidence    []ibmWordConfidence `json:"word_confidence"`
+	OverallConfidence float64             `json:"confidence"`
+	Transcript        string              `json:"transcript"`
+	Timestamps        []ibmWordTimestamp  `json:"timestamps"`
+}
+type ibmWordConfidence []interface{}
+type ibmWordTimestamp []interface{}
+
+// TranscribeWithIBM transcribes a given audio file using the IBM Watson
+// Speech To Text API
+func TranscribeWithIBM(filePath string, IBMUsername string, IBMPassword string) (*IBMResult, error) {
+	result := new(IBMResult)
+
+	url := "wss://stream.watsonplatform.net/speech-to-text/api/v1/recognize?model=en-US_BroadbandModel"
```

Comment:
Yes, this is true. I like to be explicit even when the default value is what I need. If the IBM library ever decides to change the default value, we will still be good.